src/DotNetWorker.Grpc/GrpcServiceCollectionExtensions.cs

```diff
@@ -68,9 +68,12 @@ public static IServiceCollection AddGrpc(this IServiceCollection services)
 #endif
 
             services.AddOptions<GrpcWorkerStartupOptions>()
-                .Configure<IConfiguration>((arguments, config) =>
+                .Configure<IConfiguration>((grpcWorkerStartupOption, config) =>
                 {
-                    config.Bind(arguments);
+                    grpcWorkerStartupOption.HostEndpoint = config.GetFunctionsHostGrpcUri();
+                    grpcWorkerStartupOption.RequestId = config["Functions:Worker:RequestId"] ?? config["requestId"];
+                    grpcWorkerStartupOption.WorkerId = config["Functions:Worker:WorkerId"] ?? config["workerId"];
```

Comment:
We could consider removing the fallback code after a few host releases goes out. Opened an issue to track that.

[Evaluate removing the fallback command line argument reading code(HOST, PORT)](https://github.com/Azure/azure-functions-dotnet-worker/issues/1908)